include/pcmki/pcmki_verify.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#ifndef PCMK__PCMKI_PCMKI_VERIFY__H
+#  define PCMK__PCMKI_PCMKI_VERIFY__H
+
+#include <crm/common/output_internal.h>
+#include <crm/common/scheduler.h>
+#include <pcmki/pcmki_transition.h>
+#include <crm/cib.h>                    // cib_t
+#include <pacemaker.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+/*!
+ * \internal
+ * \brief Parse a CIB file
+ *
+ * This function parses a CIB file into a CIB object
+ *
+ * \param[in]     out          Output to use for logging and printing results
+ * \param[in]     use_live_cib Whether or not to use the CIB that's running on
+ *                             the cluster as the input CIB
+ * \param[in]     xml_file     If provided, the XML file to use as the input CIB
+ * \param[in]     xml_string   If provided, the the string that constitutes the
+ *                             input CIB
+ * \param[in]     xml_stdin    Whether or not to input the CIB via stdin and pipe
+ * \param[in,out] cib_object   The resulting, parsed CIB object
+ *
+ * \return Standard Pacemaker return code
+ */
+int parse_cib_file(pcmk__output_t *out, bool use_live_cib, const char *xml_file,
+                   const char *xml_string, bool xml_stdin, xmlNodePtr *cib_object);
+
+/*!
+ * \internal
+ * \brief Verify that a CIB is error-free or output errors and warnings
+ *
+ * This high-level function essentially implements crm_verify(8). It operates
+ * on an input CIB file, which can be inputted through one of several ways. It
+ * can either write out XML-formatted output or plaintext output.
+ *
+ * \param[in,out] scheduler    Scheduler data
+ * \param[in]     out          Output to use for logging and printing results
+ * \param[in]     use_live_cib Whether or not to use the CIB that's running on
+ *                             the cluster as the input CIB
+ * \param[in]     cib_save     Where to save the verified CIB to
+ * \param[in]     verbosity    The degree of verbosity (higher is more verbose)
+ * \param[in]     cib_object   The parsed CIB object
+ *
+ * \return Standard Pacemaker return code
+ */
+int pcmk__verify(pcmk_scheduler_t *scheduler, pcmk__output_t *out,
+                 bool use_live_cib, const char *cib_save, int verbosity, xmlNode *cib_object);
```

Comment:
The internal function can take out and cib_object. crm_verify and the public function can call `parse_cib_file()` to get the cib_object to pass, and crm_verify can handle the cib_save and verbosity capabilities itself.